packages/cli/src/build.ts

```diff
@@ -1,92 +1,15 @@
-  CompilerHost,
-  CompilerOptions,
-  ParsedCommandLine,
-  Program,
-  ProjectReference,
-  System,
-} from 'typescript';
-import typescript from 'typescript';
+import type { CompilerHost, System } from 'typescript';
 
 import type { BuildType } from './build-type.js';
-import { getBuildTypeOptions } from './build-type.js';
+import { getBaseCompilerOptions, getTypeScriptConfig } from './config.js';
+import { success } from './logging.js';
+import type { BuilderOptions } from './shared.js';
 import {
-  getBaseCompilerOptions,
-  getCompilerOptions,
-  getTypeScriptConfig,
-} from './config.js';
-import { TypeScriptError } from './errors.js';
-import { getWriteFileFunction, removeDirectory } from './file-system.js';
-import { getLoggingTransformer, info } from './logging.js';
-import {
-  createProjectReferencesCompilerHost,
-  getResolvedProjectReferences,
-} from './project-references.js';
-import type { Steps } from './steps.js';
-import { executeSteps } from './steps.js';
-import type { TransformerOptions } from './transformers.js';
-import {
-  getDynamicImportExtensionTransformer,
-  getTypeImportExportTransformer,
-  getExportExtensionTransformer,
-  getImportExtensionTransformer,
-  getRequireExtensionTransformer,
-} from './transformers.js';
-import { getDefinedArray } from './utils.js';
-
-const { createProgram, getPreEmitDiagnostics, ModuleResolutionKind } =
-  typescript;
-
-type GetProgramOptions = {
-  compilerOptions: CompilerOptions;
-  projectReferences?: readonly ProjectReference[];
-  files: string[];
-  oldProgram?: Program;
-  host?: CompilerHost;
-};
-
-/**
- * Get the TypeScript program for the project. This function will create a new
- * program using the provided options and files, and perform a pre-emit
- * diagnostics check to ensure the project is valid.
- *
- * @param options - The options.
- * @param options.compilerOptions - The compiler options to use.
- * @param options.projectReferences - The project references to use.
- * @param options.files - The files to include in the program.
- * @param options.oldProgram - The old program to reuse.
- * @param options.host - The compiler host to use.
- * @returns The TypeScript program for the project.
- */
-export function getProgram({
-  compilerOptions,
-  projectReferences,
-  files,
-  oldProgram,
-  host,
-}: GetProgramOptions): Program {
-  const program = createProgram({
-    rootNames: files,
-    options: compilerOptions,
-    projectReferences,
-    oldProgram,
-    host,
-  });
-
-  // Check for pre-emit diagnostics, which includes syntax errors, type errors,
-  // and other issues that would prevent the build from starting.
-  const preEmitDiagnostics = getPreEmitDiagnostics(program);
-  if (preEmitDiagnostics.length > 0) {
-    throw new TypeScriptError(
-      'Failed to initialise the project.',
-      preEmitDiagnostics,
-    );
-  }
-
-  return program;
-}
+  getInitialProgram,
+  cleanOutputDirectory,
+  getBuildFunction,
+} from './shared.js';
```

Comment:
Same import order as in `worker-utils`:
```suggestion
  cleanOutputDirectory,
  getBuildFunction,
  getInitialProgram,
} from './shared.js';
```

Reply:
I wonder if we can enforce this with ESLint somehow.

Reply:
Yeah, I was kind of surprised to see this happen, because I thought ESLint would catch it. I could have sworn we have had this functionality in the MM config in the past. There [is an official rule for it](https://eslint.org/docs/latest/rules/sort-imports), and [it exists in v8.x](https://eslint.org/docs/v8.x/rules/sort-imports).